### DISCUSSION TITLE
Running this image (v7.0.10) from docker in Mac results in a connection refused

### DISCUSSION THREAD

**Discussion by devaljain1998** [Grokzen/docker-redis-cluster#158] [Q&A] [answered]
Hi @Grokzen,
First of all kudos to you for solving the headache problem of deploying the redis-cluster by the image that you have created! 🎉 

I am using the following command to up the redis cluster on my local machine as suggested in your README.md:
`docker run -e "IP=0.0.0.0" -p 6000-6005:7000-7005 grokzen/redis-cluster:7.0.10`.

Then I try to access the cluster using: `redis-cli -c -h localhost -p 6000`
But it gives me the following error:
```
localhost:6000> set foo bar
-> Redirected to slot [12182] located at 127.0.0.1:7002
Could not connect to Redis at 127.0.0.1:7002: Connection refused
not connected>
```
After which it disconnects.

I am running this on my Macbook (Ventura 13.4).

What is possibly wrong here?

**Comment by Grokzen** [comment]
Why are you trying to connect to port 6000? there is nothing running on that port unless you have changed around something in your dockerfile. Redis cluster runs om ports 7000-7005 and nothing else if you use the default image from me. So in your redis-cli call try to connect to the port that the error code responds with `redis-cli -c -h localhost -p 7000` and it should work unless you changed anything else internally.

**Reply by devaljain1998** [reply]
If you will look at the command where I'm `docker run -e "IP=0.0.0.0" -p 6000-6005:7000-7005 grokzen/redis-cluster:7.0.10`
I am mapping my local's 6000 to 7000 to the container's port.

My local's 7000 is occupied with another process.

**Reply by Grokzen** [reply]
Yeah, that is not going to work out :P becuase even the standard client uses the same protool as all clients where it asks the redis-cluster for what IP:port a node is at, and redis-cli do not understand that you have re-mapped your ports to something else, it will always think that the nodes is at port 7000 becuase that is what redis-cluster uses internally.

To sort this out you need to change the intiial port number for the cluster on the inside by using the envrionment variable described in this section https://github.com/Grokzen/docker-redis-cluster#change-number-of-nodes and change the INITIAL_PORT variable from 7000 to 6000 and then the entire cluster will change the ports it uses and your redis-cli should start to work as expected again

**Reply by devaljain1998** [reply]
Thanks a lot bud, for your quick response. These are extremely helpful.
I tried setting up the **INITIAL_PORT** parameter and also the running the script once again, shown in the following:
```bash
> export INITIAL_PORT=6000

> docker run -e "IP=0.0.0.0" -p 6000-6005:6000-6005 grokzen/redis-cluster:7.0.10
```

Then after running the redis-cli again I am getting the following error:
```bash
redis-cli -c -h localhost -p 6000                                                                                                                     ─╯
localhost:6000> set foo bar
Error: Server closed the connection
not connected>
```

These seemed to have solved the port mapping issue, but server is closing unexpectedly now.

**Reply by Grokzen** [reply, marked as answer]
Try with `docker run -e "INITIAL_PORT=6000" -e "IP=0.0.0.0" -p 6000-6005:6000-6005 grokzen/redis-cluster:7.0.10`

I don't think just exporting it on your host machine really pushes it down into the container unless you send it in explicitly.

If you can't connect to the node then you have to go inside the container into bash and run something like `netstat -anutp` to see what ports the service really uses because it seems that it is not setting up fully properly. Then once you have that you can telnet into that port and run commands and inspect the cluster with `CLUSTER NODES` to see what the internal state of redis-cluster says

**Reply by devaljain1998** [reply]
Thanks a lot @Grokzen!
You are extremely helpful and prompt in answering.
Also, super thankful for your valuable contributions to the community. :) 

Let's also connect over linkedin. [My ID](https://linkedin.com/in/devalsethi).
